subprojects/maven/src/main/java/org/gradle/api/publish/maven/internal/publication/DefaultMavenPublication.java

```diff
@@ -267,7 +283,38 @@ private void populateFromComponent() {
                     addDependencyConstraint(dependency, dependencyConstraints);
                 }
             }
+            if (!usageContext.getCapabilities().isEmpty()) {
+                if (unsupportedUsages == null) {
+                    unsupportedUsages = Sets.newHashSet();
+                }
+                for (Capability capability : usageContext.getCapabilities()) {
+                    unsupportedUsages.add(String.format("Declares capability %s:%s:%s", capability.getGroup(), capability.getName(), capability.getVersion()));
+                }
+            }
+        }
```

Comment:
I think we are missing validation for dependencies declared with attributes.

Reply:
Added